src/app/services/taskDetailsApi.ts

```diff
@@ -0,0 +1,29 @@
+import { TASKS_URL } from '@/constants/url';
+import { api } from './api';
+import { taskDetailsDataType } from '@/interfaces/taskDetails.type';
+
+export const taskDetailsApi = api.injectEndpoints({
```

Comment:
updateTaskDetails mutation is incorrectly using the PATCH method, when it should be using the PUT method. The PATCH method is used to update a specific resource, while the PUT method is used to replace a resource entirely. In this case, we want to replace the entire task details resource, so we should be using the PUT method.